File: 프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.cpp
#include <string>
#include <vector>

using namespace std;

vector<string> solution(vector<string> str_list) {
    vector<string> answer;
    int index = -1;

    for (int i = 0; i < str_list.size(); i++) {
        if (str_list[i] == "l" || str_list[i] == "r") {
            index = i;
            break;
        }
    }

    if (str_list[index] == "l") {
        for (int i = 0; i < index; i++) {
            answer.push_back(str_list[i]);
        }
    }
    if (str_list[index] == "r") {
        for (int i = index+1; i < str_list.size(); i++) {
            answer.push_back(str_list[i]);
        }
    }


    return answer;
}

File: 프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.swift
import Foundation

func solution(_ str_list:[String]) -> [String] {
    let leftIndex = str_list.firstIndex(of: "l") ?? 25
    let rightIndex = str_list.firstIndex(of: "r") ?? 25

    if leftIndex == 25 && rightIndex == 25 { return [] }
    
    return leftIndex < rightIndex ? Array(str_list.prefix(leftIndex)) : Array(str_list.suffix(str_list.count - rightIndex - 1))
}